doc/adrs/042_use_wildcard_collections.adoc

```diff
@@ -0,0 +1,90 @@
+= ADR-042 - Use Wildcard Collections instead of List<Object>
+
+== Context
+
+Sirius-Components has API that expose `Function<Something, List<Object>>` attributes.
+That is used especially in Descriptors (`TreeDescription`, `ListDescription`, ...)
+
+Using `List<Object>` as the return type forces implementors to copy lists returned by existing services typed with `List<Something>`, or to create lists when using a service that returns a Collection or an Iterable.
+
+Here are some alternatives:
+
+* Using `Iterable<?>`
+** Clarifies that what's returned is only intended to be iterated over, but does not make it possible to easily know the number of elements, or if there's an element at all.
+* Using `Collection<?>`
+** Better than `List<Object>`, does not impose ordering of elements, and makes it easy to check isEmpty() or size().
+* Using `List<?>`
+** Better than `List<Object>`, conveys that ordering of elements matters, and makes it easy to check isEmpty() or size().
+
+== Decision
+
```

Comment:
You mention several alternatives above, but the actrual "Decision" section does not make it clear which one was chosen.
Yes it's more or less implied by the properties of the alternatives, but it does not cost much to make the choice more explicit.